src/pages/new-chat/new-chat.html

```diff
@@ -10,9 +10,15 @@
 
 <ion-content class="new-chat">
   <ion-list class="users">
-    <button ion-item *ngFor="let user of users | async" class="user" (click)="addChat(user)">
+    <button ion-item *ngFor="let user of users | async | search:searchPattern:'profile.name'" class="user" (click)="addChat(user)">
       <img class="user-picture" [src]="user.profile.picture">
       <h2 class="user-name">{{user.profile.name}}</h2>
     </button>
   </ion-list>
 </ion-content>
+
+<ion-footer>
+  <ion-toolbar color="whatsapp" position="bottom">
+    <ion-searchbar color="whatsapp" [(ngModel)]="searchPattern"></ion-searchbar>
```

Comment:
`ngModel` might be an overkill since the pattern can be changed only from the user's input and not from the code. 
I think that it's best to register to an change event only (one way), and then use the new value from the event to `next` a RxJS `Subject` that handles the data changes. 
This way you use less dirty-check of Angular 2 and more specific reactivity changes.